### PR TITLE
feat(appsync): support appsync functions for pipelineConfig

### LINKED ISSUE #9092: [aws-appsync]: implementation for appsync::functionConfiguration for Pipeline Resolvers
Current implementation doesn't offer customers a smooth ability to create AppSync Functions for Pipeline Resolvers. 

Supersedes #6923 

### Use Case

If users want to utilize AppSync's graphQL functions that are serializable and functionable.

### Proposed Solution

Workaround currently would be to make `Lambda Functions` but these are expensive and will costly for customers. 

Possible solution could be to create an class that creates AppSync Functions

```ts
interface GraphQLFunctionProps {
    dataSourceName: string,
   description: string,
   functionVersion: string,
   name: string,
   requestMappingTemplate?: MappingTemplate,
   responseMappingTemplate?: MappingTemplate,
}

class GraphQLFunction extends Construct { ... }

class GraphQLApi extends Construct {
...
  createGraphQLFunction(...);
...
}
```

### Other

The `Function` that can be added to the `pipelineConfig` is not a `Lambda Function`. Instead, it is `AppSync::Function`. 

See documentation on `pipelineConfig` [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appsync-resolver-pipelineconfig.html).

See documentation on `functionConfiguration` [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-functionconfiguration.html).

See documentation on `appsync:CreateFunction` [here](https://docs.aws.amazon.com/appsync/latest/APIReference/API_CreateFunction.html).

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by BryanPan342**
Support AppSync Function by exposing Function Configuration. 

**BREAKING CHANGES**: `Resolver.pipelineConfig` no longer supports `string[]`
- **AppSync**: `Resolver.pipelineConfig` no longer supports `string[]`, instead use `AppsyncFunction`
- **AppSync**: Resolvers are scoped from `GraphqlApi` instead of its `Data Source`, this means that the **logical id** of resolvers will change!

Fixes #9092

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/workflows/closed-issue-message.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Closed Issue Message
 on:
-    issues:
-       types: [closed]
+  issues:
+    types: [closed]
```
Comment (shivlaks) [inline .github/workflows/closed-issue-message.yml:4]: dont think you intended to bring this change into this PR - leftover merge artifact?

Comment (BryanPan342) [reply .github/workflows/closed-issue-message.yml]: ah yes thanks for the check

**Comment by bryan-hunter** [comment]
Fellow Bryan here :)

any thoughts on what this needs to get merged and released?

**Comment by BryanPan342** [comment]
@bryan-hunter Hi Bryan! 

I think the meat of the work is done, just waiting for a review at this point! It's a big PR, which is totally on me. But I saw some discrepancies with the implementation of `Appsync.Resolvers` so I ended up fixing it in this PR as well.

@MrArnoldPalmer would it be easier if I split this PR with the breaking change refactor and then the new feature of AppSync Functions?

**Comment by davekiss** [comment]
also looking forward to this release!

**Comment by julianmee** [comment]
Would be very nice to have this feature!

**Review by MrArnoldPalmer — APPROVED** [review]
@BryanPan342 good 2 go here once conflicts are resolved. Took me a bit to test this out.

**Comment by BryanPan342** [comment]
> @BryanPan342 good 2 go here once conflicts are resolved. Took me a bit to test this out.

woooo thanks @MrArnoldPalmer 😊 yeah sorry this should have been two PR's

**Pull request merged**